pandas/core/window/rolling.py

```diff
@@ -1708,9 +1777,22 @@ def _validate(self):
 
         elif isinstance(self.window, BaseIndexer):
             # Passed BaseIndexer subclass should handle all other rolling kwargs
-            return
+            pass
         elif not is_integer(self.window) or self.window < 0:
             raise ValueError("window must be an integer 0 or greater")
+        # GH 15354:
+        # validate window indexer parameters do not raise in get_window_bounds
+        # this cannot be done in BaseWindow._validate because there _get_window_indexer
+        # would erroneously create a fixed window given a window argument like "1s" due
+        # to _win_freq_i8 not being set
+        indexer = self._get_window_indexer()
```

Comment:
How about just

```
if self.step is not None:
    raise NotImplementedError(...)
```

Reply:
I believe this would be too restrictive and would fail a numeric step that is actually valid with fixed-length window indexers.